src/Auth.ts

```diff
@@ -48,30 +51,30 @@ export class Class {
 		);
 	}
 
-	authenticate( username:string, password:string ):Promise<void>;
-	authenticate( authenticationToken:AuthenticationToken ):Promise<void>;
-	authenticate( usernameOrToken:any, password:string = null ):Promise<void> {
-		return new Promise<void>( ( resolve:( result:any ) => void, reject:( reject:Error ) => void ) => {
-			if( ! usernameOrToken ) throw new Errors.IllegalArgumentError( "Either a username or an authenticationToken are required." );
-
-			let authenticationToken:AuthenticationToken;
-			if( Utils.isString( usernameOrToken ) ) {
-				let username:string = usernameOrToken;
-				if( ! password ) throw new Errors.IllegalArgumentError( "A password is required when providing a username." );
-				authenticationToken = new UsernameAndPasswordToken( username, password );
-			} else {
-				authenticationToken = usernameOrToken;
-			}
-
-			if( this.authenticator ) this.clearAuthentication();
-
-			this.authenticator = this.getAuthenticator( authenticationToken );
+	authenticate( username:string, password:string ):Promise<Credentials> {
+		return this.authenticateUsing( "TOKEN", username, password );
+	}
 
-			resolve( this.authenticator.authenticate( authenticationToken ) );
-		});
+	authenticateUsing( method:"BASIC", username:string, password:string ):Promise<UsernameAndPasswordCredentials>;
+	authenticateUsing( method:"TOKEN", username:string, password:string ):Promise<TokenCredentials>;
```

Comment:
Please add the following signature:
authenticateUsing( method:"TOKEN", token:Token.Class ):Promise;